hack/update-vendor.sh

```diff
@@ -180,210 +142,98 @@ function add_generated_comments() {
   go mod edit -fmt
 }
 
+function add_staging_replace_directives() {
```

Comment:
this is moved from `Phase 2` and made a function we can apply to the root and staging modules, rather than our previous approach of messing with the root go.mod, copying it to all staging dirs, then fixing it up for the staging dirs

Reply:
sounds good!